pkg/asset/installconfig/gcp/validation.go

```diff
@@ -15,11 +15,28 @@ import (
 func Validate(client API, ic *types.InstallConfig) error {
 	allErrs := field.ErrorList{}
 
+	allErrs = append(allErrs, validateProject(client, ic, field.NewPath("platform").Child("gcp"))...)
 	allErrs = append(allErrs, validateNetworks(client, ic, field.NewPath("platform").Child("gcp"))...)
 
 	return allErrs.ToAggregate()
 }
 
+func validateProject(client API, ic *types.InstallConfig, fieldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if ic.GCP.ProjectID != "" {
+		projects, err := client.GetProjects(context.TODO())
+		if err != nil {
+			return append(allErrs, field.Invalid(fieldPath.Child("project"), ic.GCP.ProjectID, "failed to retrieve project IDs"))
```

Comment:
Abhinav's point in https://github.com/openshift/installer/pull/3484#discussion_r412557760 was that `field.Invalid` -> `field.InternalError`. `field.Invalid` means the user's input was bad, but in this case there was an error when calling the gcloud API.

You could fix the preexisting code in a separate commit as well.

Reply:
> Abhinav's point in #3484 (comment) was that field.Invalid -> field.InternalError.

I get it now thanks.

> You could fix the preexisting code in a separate commit as well.

Got it. Will follow that next time.

Reply:
> You could fix the preexisting code in a separate commit as well.

> Got it. Will follow that next time.

No I meant that the earlier code (that I wrote) has the same issue and you could fix it in a separate commit.